frontend/pages/ManageControlsPage/Secrets/Secrets.tsx

```diff
@@ -37,6 +45,16 @@ const Secrets = () => {
 
   const queryClient = useQueryClient();
 
+  const {
+    isGlobalAdmin,
+    isTeamAdmin,
+    isGlobalMaintainer,
+    isTeamMaintainer,
+  } = useContext(AppContext);
+
+  const canEdit =
+    isGlobalAdmin || isTeamAdmin || isGlobalMaintainer || isTeamMaintainer;
+
```

Comment:
Controls page isn't actually visible to Observers, but in case that changes (or we move this section), the spec says only maintainer+ should be able modify secrets.